plugins/opusCodec/src/OpusEncoder.cpp

```diff
@@ -0,0 +1,269 @@
+        qCCritical(encoder) << "Failed to initialize Opus encoder: " << error_to_string(error);
+        _encoder = nullptr;
+        return;
+    }
+
+    setBitrate(DEFAULT_BITRATE);
+    setComplexity(DEFAULT_COMPLEXITY);
+    setApplication(DEFAULT_APPLICATION);
+    setSignal(DEFAULT_SIGNAL);
+
+    qCDebug(encoder) << "Opus encoder initialized, sampleRate = " << sampleRate << "; numChannels = " << numChannels;
+}
+
+AthenaOpusEncoder::~AthenaOpusEncoder()
+{
+    opus_encoder_destroy(_encoder);
+}
+
+
+
+void AthenaOpusEncoder::encode(const QByteArray& decodedBuffer, QByteArray& encodedBuffer) {
+
+    PerformanceTimer perfTimer("AthenaOpusEncoder::encode");
+    assert(_encoder);
+
+    encodedBuffer.resize( decodedBuffer.size() );
+    int frame_size = decodedBuffer.length()/ _opus_channels / static_cast<int>(sizeof(opus_int16));
+
+    int bytes = opus_encode(_encoder, reinterpret_cast<const opus_int16*>(decodedBuffer.constData()), frame_size, reinterpret_cast<unsigned char*>(encodedBuffer.data()), encodedBuffer.size() );
+
+    if ( bytes >= 0 ) {
+        //qCDebug(encoder) << "Encoded " << decodedBuffer.length() << " bytes into " << bytes << " opus bytes";
+        encodedBuffer.resize(bytes);
+    } else {
+        encodedBuffer.resize(0);
+
+        qCWarning(encoder) << "Error when encoding " << decodedBuffer.length() << " bytes of audio: " << error_to_string(bytes);
+    }
+
+}
+
+int AthenaOpusEncoder::getComplexity() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_COMPLEXITY(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setComplexity(int complexity)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_COMPLEXITY(complexity));
+
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting complexity to " << complexity << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getBitrate() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_BITRATE(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setBitrate(int bitrate)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_BITRATE(bitrate));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting bitrate to " << bitrate << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getVBR() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setVBR(int vbr)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_VBR(vbr));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting VBR to " << vbr << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getVBRConstraint() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR_CONSTRAINT(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setVBRConstraint(int vbr_const)
+{
```

Comment:
Please put { on same line as function name  CS 4.1.4 https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#414-method-definitions-should-have-the-following-form